.github/workflows/ci.yml

```diff
@@ -139,4 +162,142 @@ jobs:
         run: make test ARGS=-V
 
+      - name: Archive CodeCompass binaries
+        uses: actions/upload-artifact@v2
+        with:
+          name: codecompass-${{ matrix.os }}-${{ matrix.db }}-bin
+          path: ${{github.workspace}}/${{ matrix.os }}/${{ matrix.db }}/install
+
+  parse:
+    needs: build
+    strategy:
+      matrix:
+        db: [ postgresql, sqlite3 ]
+        os: [ ubuntu-20.04, ubuntu-18.04 ]
+      fail-fast: false
 
+    runs-on: ${{ matrix.os }}
+
+    # Service containers to run with `runner-job`
+    services:
+      # Label used to access the service container
+      postgres:
+        # Docker Hub image
+        image: postgres
+        # Provide the password for postgres
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+        # Set health checks to wait until postgres has started
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          # Maps tcp port 5432 on service container to the host
+          - 5432:5432
+
+    steps:
+      # Ubuntu 18.04 commands
+      - name: Install required packages Ubuntu 18
+        if: ${{ matrix.os == 'ubuntu-18.04' }}
+        run: sudo apt-get install -y git cmake make g++ gcc-7-plugin-dev libboost-all-dev llvm-10-dev clang-10 libclang-10-dev default-jdk libssl1.0-dev libgraphviz-dev libmagic-dev libgit2-dev ctags libgtest-dev npm
+
+      - name: Install Postgresql Ubuntu 18
+        if: ${{ matrix.os == 'ubuntu-18.04' && matrix.db == 'postgresql' }}
+        run: sudo apt-get install postgresql-server-dev-10
+
+      - name: Install SQLite3 Ubuntu 18
+        if: ${{ matrix.os == 'ubuntu-18.04' && matrix.db == 'sqlite3' }}
+        run: sudo apt-get install libsqlite3-dev
+
+      - name: Cache ODB Ubuntu 18
+        id: odb-cache
+        if: ${{ matrix.os == 'ubuntu-18.04' }}
+        uses: actions/cache@v2
+        with:
+          path: ~/odb_install
+          key: ${{ matrix.db }}-odb-v2
+
+      - name: Cache Thrift Ubuntu 18
+        id: thrift-cache
+        if: ${{ matrix.os == 'ubuntu-18.04' }}
+        uses: actions/cache@v2
+        with:
+          path: ~/thrift_install
+          key: ${{ matrix.db }}-thrift-v2
+
+      - name: Export environment variables Ubuntu 18
+        if: ${{ matrix.os == 'ubuntu-18.04' }}
+        run: |
+          echo "LD_LIBRARY_PATH=$HOME/thrift_install/lib:$HOME/odb_install/lib:$CMAKE_PREFIX_PATH" >> $GITHUB_ENV
+          echo "PATH=$HOME/thrift_install/bin:$HOME/odb_install/bin:$PATH" >> $GITHUB_ENV
+
+      # Ubuntu 20.04 commands
+      - name: Install required packages Ubuntu 20
+        if: ${{ matrix.os == 'ubuntu-20.04' }}
+        run: sudo apt-get install -y git cmake make g++ libboost-all-dev llvm-10-dev clang-10 libclang-10-dev odb libodb-dev thrift-compiler libthrift-dev default-jdk libssl-dev libgraphviz-dev libmagic-dev libgit2-dev ctags libgtest-dev npm
```

Comment:
Do we need all things here? The runtime dependencies of CodeCompass are not as extensive as the build type dependencies:
https://github.com/Ericsson/CodeCompass/blob/master/docker/runtime/Dockerfile#L44-L58

Reply:
I erased some unnecessary dependencies.

Reply:
I still see a bunch of unnecessary development packages, which are not required at runtime, based on the corresponding Docker environment:
https://github.com/Ericsson/CodeCompass/blob/master/docker/runtime/Dockerfile#L57

I suggest the following:
- Replace `libboost-all-dev`
With `libboost-filesystem-dev libboost-log-dev libboost-program-options-dev `

- Replace `llvm-10-dev clang-10 libclang-10-dev`
With `llvm-10`

- Replace `default-jdk`
With `default-jre`

- Replace `libssl-dev`
With `libssl1.1`

- Add `libgvc6` and `libmagic-dev` for GraphViz and magic.